(entorno_jevg) ttm@raspberrypi:~/TT2_Photovoltaic_Roof_Cleaner $ sudo /home/ttm/TT2_Photovoltaic_Roof_Cleaner/entorno_jevg/bin/python /home/ttm/TT2_Photovoltaic_Roof_Cleaner/SistemaControlComunicacion/main.py
/home/ttm/TT2_Photovoltaic_Roof_Cleaner/entorno_jevg/lib/python3.11/site-packages/gpiozero/devices.py:300: PinFactoryFallback: Falling back from lgpio: No module named 'lgpio'
  warnings.warn(
/home/ttm/TT2_Photovoltaic_Roof_Cleaner/entorno_jevg/lib/python3.11/site-packages/gpiozero/devices.py:300: PinFactoryFallback: Falling back from rpigpio: No module named 'RPi'
  warnings.warn(
/home/ttm/TT2_Photovoltaic_Roof_Cleaner/entorno_jevg/lib/python3.11/site-packages/gpiozero/devices.py:300: PinFactoryFallback: Falling back from pigpio: No module named 'pigpio'
  warnings.warn(
/home/ttm/TT2_Photovoltaic_Roof_Cleaner/entorno_jevg/lib/python3.11/site-packages/gpiozero/devices.py:300: PinFactoryFallback: Falling back from native: unable to determine gpio base
  warnings.warn(
Traceback (most recent call last):
  File "/home/ttm/TT2_Photovoltaic_Roof_Cleaner/SistemaControlComunicacion/main.py", line 3, in <module>
    import callbacks
  File "/home/ttm/TT2_Photovoltaic_Roof_Cleaner/SistemaControlComunicacion/callbacks.py", line 3, in <module>
    from functions import move_forward, move_backward, move_left, move_right, rotate_right, rotate_left
  File "/home/ttm/TT2_Photovoltaic_Roof_Cleaner/SistemaControlComunicacion/functions.py", line 47, in <module>
    motor1 = StepperMotor(23, 24, 25)
             ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ttm/TT2_Photovoltaic_Roof_Cleaner/SistemaControlComunicacion/functions.py", line 19, in __init__
    self.PUL = OutputDevice(pul_pin)
               ^^^^^^^^^^^^^^^^^^^^^
  File "/home/ttm/TT2_Photovoltaic_Roof_Cleaner/entorno_jevg/lib/python3.11/site-packages/gpiozero/devices.py", line 108, in __call__
    self = super().__call__(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ttm/TT2_Photovoltaic_Roof_Cleaner/entorno_jevg/lib/python3.11/site-packages/gpiozero/output_devices.py", line 74, in __init__
    super().__init__(pin, pin_factory=pin_factory)
  File "/home/ttm/TT2_Photovoltaic_Roof_Cleaner/entorno_jevg/lib/python3.11/site-packages/gpiozero/mixins.py", line 75, in __init__
    super().__init__(*args, **kwargs)
  File "/home/ttm/TT2_Photovoltaic_Roof_Cleaner/entorno_jevg/lib/python3.11/site-packages/gpiozero/devices.py", line 544, in __init__
    super().__init__(pin_factory=pin_factory)
  File "/home/ttm/TT2_Photovoltaic_Roof_Cleaner/entorno_jevg/lib/python3.11/site-packages/gpiozero/devices.py", line 245, in __init__
    Device.ensure_pin_factory()
  File "/home/ttm/TT2_Photovoltaic_Roof_Cleaner/entorno_jevg/lib/python3.11/site-packages/gpiozero/devices.py", line 270, in ensure_pin_factory
    Device.pin_factory = Device._default_pin_factory()
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/ttm/TT2_Photovoltaic_Roof_Cleaner/entorno_jevg/lib/python3.11/site-packages/gpiozero/devices.py", line 302, in _default_pin_factory
    raise BadPinFactory('Unable to load any default pin factory!')
gpiozero.exc.BadPinFactory: Unable to load any default pin factory!